electron/custom_install_steps.nsh

```diff
@@ -26,32 +43,63 @@
     nsExec::ExecToStack 'add_tap_device.bat i386'
   ${EndIf}
 
-  
   Pop $0
   Pop $1
   StrCmp $0 0 installservice
-  MessageBox MB_OK "Sorry, we could not configure your system to connect to Outline. Please try running the installer again.$\n$\nIf you still cannot install Outline, please get in touch with us and let us know that the TAP device failed to install with this error: $1"
-  ; TODO: Abort gracefully, i.e. uninstall, before exiting.
-  Quit
+  MessageBox MB_OK "Sorry, we could not configure your system to connect to Outline. Please try \
+    running the installer again. If you still cannot install Outline, please get in \
+    touch with us and let us know that the TAP device failed to install with this error:$\n$\n$1"
 
-  installservice:
+  ; Submit a Sentry error event.
+  ;
+  ; This will get bundled into an issue named "could not install TAP device" with the following
+  ; attributes:
+  ;  - a single breadcrumb containing the output of add_tap_device.bat
+  ;  - Windows version, as a tag named "os" with a value identical in most cases to what the
+  ;    JavaScript Sentry client produces, e.g. "Windows 10.0.17134"
+  ;  - client version
+  ;
+  ; Note:
+  ;  - Sentry won't accept a breadcrumbs without a timestamp; fortunately, it accepts obviously
+  ;    bogus values so we don't have to fetch the real time.
+  ;  - Because nsExec::ExecToStack yields "NSIS strings" strings suitable for inclusion in, for
+  ;    example, a MessageBox, e.g. "device not found$\ncommand failed", we must convert it to a
+  ;    string that Sentry will like *and* can fit on one line, e.g.
+  ;    "device not found\ncommand failed"; fortunately, StrFunc.nsh's StrNSISToIO does precisely
+  ;    this.
 
-  ; Stop the service so we can extract the updated executable.
-  ; Note that ordinarily the uninstall steps, below, also stops the service.
-  ; This is for (really) old clients that don't have the uninstall step.
-  nsExec::Exec "net stop OutlineService"
+  ; http://nsis.sourceforge.net/Docs/StrFunc/StrFunc.txt
+  Var /GLOBAL FAILURE_MESSAGE
+  ${StrNSISToIO} $FAILURE_MESSAGE $1
 
-  File "${PROJECT_DIR}\OutlineService.exe"
-  File "${PROJECT_DIR}\Newtonsoft.Json.dll"
-  File "${PROJECT_DIR}\electron\install_windows_service.bat"
+  ${WinVerGetMajor} $R0
+  ${WinVerGetMinor} $R1
+  ${WinVerGetBuild} $R2
+
+  ; http://nsis.sourceforge.net/Inetc_plug-in#post
+  inetc::post '{\
+    "message":"could not install TAP device",\
+    "release":"${RELEASE}",\
+    "tags":[\
+      ["os", "Windows $R0.$R1.$R2"]\
+    ],\
+    "breadcrumbs":[\
+      {"timestamp":1, "message":"$FAILURE_MESSAGE"}\
+    ]\
+  }' /TOSTACK ${SENTRY_DSN} /END

```

Comment:
Could you add a comment about `$SENTRY_DSN` and `$RELEASE` being globally defined?